**Guerois Lab.** (I2BC), IBITECS, CEA, CNRS, Univ Paris-Sud, Universit√© Paris-Saclay,    (Raphael.GUEROIS@cea.fr)

The file **GueroisAndreani.csv** contains the list of entries from the benchmark v2 as updated in April 2021 with the values of the scores calculated using different methods used in the InterEvDock server.

Scores used in InterEvDock server such as IESh or SPPh scores evaluate the likelihood of interface models generated by rigid-body docking using coevolutionary information. InterEvDock scores were not trained for the discrimination between biological and non-biological units.

Description of the scores in the headers 

* IES : InterEvScore is a coarse-grained scoring function calculated at the residue level assessing the likelihood of an interface using coevolution information (10.1093/bioinformatics/btt260). It involves a combination of pairwise and 3-body statistical contact potentials. It was calculated from multiple sequence alignments containing at most 100 sequences. 

* SPP : SOAP-PP is an atomic statistical potential trained to discriminate correct from incorrect interfaces generated by rigid-body docking models (developed in A. Sali's group doi:10.1093/bioinformatics/btt560). It doesn't take evolutionary information into account.

* IESh-10seq : Is a novel implementation of InterEvScore using coevolutionary information at atomic level to improve near-native discrimination rates in scoring. We make use of explicitely modelled homologous decoys (interologs) to enrich scoring of each query decoy by averaging the scores over the query and its homologs (doi:10.1101/2020.10.26.355073). IESh_10seq was calculated using 10 interologs distributed over the sequence alignment.

* SPPh-10seq : Is a novel implementation of SOAP-PP using coevolutionary information at atomic level to improve near-native discrimination rates in scoring. We make use of explicitly modelled homologous decoys (interologs) to enrich scoring of each query decoy by averaging the SPP scores over the query and its homologs (doi:10.1101/2020.10.26.355073). SPPh_10seq was calculated using 10 interologs distributed over the sequence alignment.

* IESh-10lastseq : Same as IESh_10seq, except that we used the 10 last sequences of the alignment to increase the diversity and potentially discriminate better biological from non biological units.

* SPPh-10lastseq : Same as IESh_10seq, except that we used the 10 last sequences of the alignment to increase the diversity and potentially discriminate better biological from non biological units.

* IES-normNseqs : IES score normalized by the number of sequences in the coalignment.

* IESh-10seq_normsize : the IESh_10seq score normalized by the size of the interface.

* SPPh-10seq_normsize : the SPPh_10seq score normalized by the size of the interface.

* IESh-10lastseq-normsize : the IESh_10lastseq score normalized by the size of the interface.

* SPPh-10lastseq-normsize : the SPPh_10lastseq score normalized by the size of the interface.


Remarks:

In case a score could not be calculated for a specific pdb, a 'nan' value was added instead. For those cases there might be issues with the format of the input pdb that we may need to reformat prior to running the calculation but we did not yet implement this additional prior treatment. Could be done if the scores exhibit interesting trends. 
The scores mentioned above can be calculated for a specific interface using the singularity image provided in http://biodev.cea.fr/interevol/interevdata/data/interevdata_tools.sif (see usage in http://biodev.cea.fr/interevol/interevdata/singularity.html).
